mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -413,22 +407,25 @@ def init_inert_gas(self):
 
     def init_solver(self):
         logger.info("### Init Diode Solver Setup ###")
-        if self.dim == 1:
-            raise NotImplementedError(
-                "1D solving is not yet implemented in mewarpx")
-            # self.solver = poisson1d.PoissonSolver1D()
-        elif self.dim in [2, 3]:
-            if self.DIRECT_SOLVER:
+        if self.DIRECT_SOLVER:
+            if self.dim == 1:
+                raise NotImplementedError(
+                    "Direct 1D solving is not yet implemented in mewarpx")
+                # self.solver = poisson1d.PoissonSolver1D()
+            elif self.dim == 2:
+                if self.rz:
+                    raise NotImplementedError(
+                        "Direct RZ solving is not yet implemented in mewarpx")
                 self.solver = poisson_pseudo_1d.PoissonSolverPseudo1D(
                     grid=mwxrun.grid
                 )
-            else:
-                self.solver = picmi.ElectrostaticSolver(
-                    grid=mwxrun.grid,
-                    method='Multigrid',
-                    required_precision=1e-6,
-                    maximum_iterations=10000
-                )
+        else:
+            self.solver = picmi.ElectrostaticSolver(
+                grid=mwxrun.grid,
+                method='Multigrid',
+                required_precision=1e-6,
+                maximum_iterations=10000
+            )
```

Comment:
Would it be useful to make another PR to allow `diode_setup` to set an absolute error for the electrostatic solver?

Reply:
Yeah absolutely, if you think there is a need for it go ahead.